src/main/java/uk/gov/companieshouse/acsp/manage/users/service/UsersService.java

```diff
@@ -56,6 +56,19 @@ public User fetchUserDetails( final String userId ) {
         return createFetchUserDetailsRequest(userId).get();
     }
 
+  public boolean doesUserExist(final String userId) {
```

Comment:
In most scenarios, when we call fetchUserDetails to retrieve the user, we want to do more than just check if they exist e.g. we might want to grab their displayName. It's pretty rare where we only want to know if they exist, so this method has limited reuse potential. It might be better to just call fetchUserDetails in the controller, catch the exception and throw a BadRequestRuntime exception in the catch body

Reply:
I had split this method out just for general readability and so it was a single unit of code that could be tested.

I guess this just read better to me:

```java
if (!usersService.doesUserExist(inviteeUserId)) {
```

https://github.com/companieshouse/acsp-manage-users-api/blob/c18a21e5549fa5975402df6a1ccdf0301b0e89e2/src/main/java/uk/gov/companieshouse/acsp/manage/users/service/UsersService.java#L39 Maybe an exception might be better to bubble up to give API users more information about why their request was bad?

Reply:
Kartheek said he wants us to return imprecise information about why the endpoint failed, to add a layer of obfuscation/security - hence "Please check the request and try again" is what we largely return.